bouncer/src/repo/core/model/bson/repo_bson_factory.cpp

```diff
@@ -796,12 +803,33 @@ RepoUnityAssets RepoBSONFactory::makeRepoBundleAssets(
 		builder.append(REPO_LABEL_MODEL, model);
 
 	if (offset.size())
-		builder.appendArray(REPO_UNITY_ASSETS_LABEL_OFFSET, offset);
+		builder.appendArray(REPO_ASSETS_LABEL_OFFSET, offset);
 
 	if (repoJsonFiles.size())
-		builder.appendArray(REPO_UNITY_ASSETS_LABEL_JSONFILES, repoJsonFiles);
+		builder.appendArray(REPO_ASSETS_LABEL_JSONFILES, repoJsonFiles);
+
+	// Metadata is provided in an array with the same indexing as asset names.
+	// The metadata schema uses the object (instead of array) encoding of
+	// vectors, so that the type & entire array is a fixed size and cam be
+	// pre-allocated.
+
+	std::vector<RepoBSON> metadataNodes;
+	for (auto& meta : metadata)
+	{
+		RepoBSONBuilder metadataBuilder;
+		metadataBuilder.append(REPO_ASSETS_LABEL_NUMVERTICES, (unsigned int)meta.numVertices);
+		metadataBuilder.append(REPO_ASSETS_LABEL_NUMFACES, (unsigned int)meta.numFaces);
+		metadataBuilder.append(REPO_ASSETS_LABEL_NUMUVCHANNELS, (unsigned int)meta.numUVChannels);
+		metadataBuilder.append(REPO_ASSETS_LABEL_PRIMITIVE, (unsigned int)meta.primitive);
+		metadataBuilder.appendVector3d(REPO_ASSETS_LABEL_MIN, meta.min);
+		metadataBuilder.appendVector3d(REPO_ASSETS_LABEL_MIN, meta.max);
+		metadataNodes.push_back(metadataBuilder.obj());
+	}
+
+	if (metadataNodes.size())
```

Comment:
@sebjf  started reading https://github.com/3drepo/3drepounity/pull/534/files and this thought came up...

Since this is stored as a bson object in mongo, are we certain that by adding the metadata nodes here, we dont' risk pushing the size of the bson beyond 16MB (which is the limit per document on the database)?

Reply:
Hi @carmenfan, good point! I have worked this out...

Using the objsize() method of the assets list document returned by `makeRepoBundleAssets` (which I assume is an appropriate measure because that's what we used to do the bin mapping before!), the document size increases just sub-linearly with the number of entries, starting at 263 bytes per entry.

So, we could fit 63,791 assets into a document before we run out of space.

If each supermesh had the maximum number of submeshes (5000), that would mean a container could have 318,958,479 elements before we hit the limit on this document.

I have added a unit test that is pretty much at this limit, so if the schema changes we'll get a warning to consider this again.